### PR TITLE
Add "requiredannotations"

### PR DESCRIPTION
**Pull request by gabops**
Hello,

This PR adds policy and resources for enforcing the declaration of certain set of annotations on an k8s object. I got the idea after reading [this article](https://kubernetes.io/blog/2021/04/20/annotating-k8s-for-humans/) from the official Kubernetes blog.

Thank you in advance!

Regards

### CODE REVIEW

File: library/general/requiredannotations/template.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                  key:
+                    type: string
+                  allowedRegex:
+                    type: string
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package k8srequiredannotations
+
+        violation[{"msg": msg, "details": {"missing_annotation(s)": missing}}] {
```
Comment (maxsmythe) [inline library/general/requiredannotations/template.yaml:32]: nit: `details` is meant to be machine readable, no need for `(s)`

**Review by maxsmythe — COMMENTED** [review]
Thank you for your PR!

This LGTM after 1 nit.

**Review by maxsmythe — APPROVED** [review]
LGTM!

@sozercan @shomron @ritazh LGTY?

**Review by sozercan — APPROVED** [review]
LGTM

**Pull request merged**